test/burn/org/apache/cassandra/utils/memory/LongBufferPoolTest.java

```diff
@@ -181,7 +222,7 @@ void checkpoint()
                 void testOne() throws Exception
                 {
 
-                    long currentTargetSize = rand.nextInt(poolSize / 1024) == 0 ? 0 : targetSize;
+                    long currentTargetSize = (rand.nextInt(poolSize / 1024) == 0 || freedAllMemory[threadIdx].get() == false) ? 0 : targetSize;
```

Comment:
```suggestion
                    long currentTargetSize = (rand.nextInt(poolSize / 1024) == 0 || !freedAllMemory[threadIdx].get()) ? 0 : targetSize;
```

This is the canonical way in Java.

Reply:
thx